components/ProfileDropdownMenu.tsx

```diff
@@ -0,0 +1,125 @@
+    // 'A' stands for Anonymous, in case user did not put in full name
+    username,
+    firstName: fullname.split(' ')[0] || 'A',
+    lastName: fullname.split(' ')[1] || ' ',
+    discordUserId: _.get(data, 'userInfo.user.discordUserId', ''),
+    discordUsername: _.get(data, 'userInfo.user.discordUsername', ''),
+    discordAvatarUrl: _.get(data, 'userInfo.user.discordAvatarUrl', '')
+  }
+
+  const dropdownAdminMenuItems = [
+    { title: 'Lessons', path: ADMIN_PATH + '/lessons' },
+    { title: 'Users', path: ADMIN_PATH + '/users' },
+    { title: 'Alerts', path: ADMIN_PATH + '/alerts' }
+  ]
+
+  const isActive = (path: string) => {
+    if (path === location) return `${styles['active']}`
+    return ''
+  }
+
+  const adminDropdownMenu = dropdownAdminMenuItems.map(({ title, path }) => (
+    <Dropdown.Item
+      className={`${styles['dropdown-item']}`}
+      href={path}
+      key={title}
+      bsPrefix={isActive(path)}
+    >
+      {title}
+    </Dropdown.Item>
+  ))
+
+  const buttonMenuToggle = React.forwardRef<
+    HTMLDivElement,
+    React.ComponentPropsWithoutRef<'div'>
+  >((props, ref) => <div ref={ref} {...props}></div>)
+
+  return (
+    <Dropdown>
+      <div className="d-none d-lg-block">
+        <Dropdown.Toggle
+          as={buttonMenuToggle}
+          className={`${styles['nav-user-toggle']} `}
+          id="user_nav_toggle"
+        >
+          <UserInfoImage user={userInfo} className={`${styles['user-icon']}`} />
+          <span>{_.capitalize(username)}</span>
+          <ChevronDownIcon size={16} />
+        </Dropdown.Toggle>
+        <Dropdown.Menu className={`${styles['dropdown-menu']}`} align="end">
+          {isAdmin && (
+            <>
+              <Dropdown.Header>ADMIN</Dropdown.Header>
+              {adminDropdownMenu}
+              <Dropdown.Divider />
+            </>
+          )}
+          <Dropdown.Item
+            className={`${styles['dropdown-item']} `}
+            bsPrefix={isActive(userProfilePath)}
+            href={userProfilePath}
+          >
+            Profile
+          </Dropdown.Item>
+          <Dropdown.Item className={`${styles['dropdown-item']}`}>
+            <LogoutContainer>Logout</LogoutContainer>
+          </Dropdown.Item>
+        </Dropdown.Menu>
+      </div>
+      <div className="d-lg-none">
+        {adminDropdownMenu}
+        <Dropdown.Item
+          className={`${styles['dropdown-item']}`}
+          bsPrefix={isActive(userProfilePath)}
+          href={userProfilePath}
+        >
+          Profile
+        </Dropdown.Item>
+        <Dropdown.Item className={`${styles['dropdown-item']} `}>
+          <LogoutContainer>
+            <div className={`${styles['light-button']} d-inline`}>
+              <Button border ml="2" type="light">
+                Logout
+              </Button>
+            </div>
+          </LogoutContainer>
+        </Dropdown.Item>
+      </div>
+    </Dropdown>
+  )
+}
+
+export default ProfileDropdownMenu
```

Comment:
Lets make a story for this component. This will be useful in the future incase anyone needs to use this component again in the future.

Profile component stories are here
https://github.com/garageScript/c0d3-app/tree/master/stories/profile

Reply:
This PR has been open for awhile so I don't wanna make this a blocker, but please do follow up with making a story for this in another PR since it is a part of engineering practices

Reply:
This component isn't reusable, so story is not needed

Reply:
Marked as resolved because there is no need for a story, the component isn't reusable.